packages/zoe/test/unitTests/contracts/test-zoeHelpersWZcf.js

```diff
@@ -0,0 +1,129 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import '@agoric/install-ses';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import test from 'ava';
+
+import { E } from '@agoric/eventual-send';
+import bundleSource from '@agoric/bundle-source';
+
+// noinspection ES6PreferShortImport
+import { makeZoe } from '../../../src/zoeService/zoe';
+import { setup } from '../setupBasicMints';
+import fakeVatAdmin from './fakeVatAdmin';
+import { swap } from '../../../src/contractSupport';
+import { assertPayoutAmount } from '../../zoeTestHelpers';
+
+const contractRoot = `${__dirname}/zcfTesterContract`;
+
+test(`zoeHelper with zcf - swap`, async t => {
+  const {
+    moolaIssuer,
+    moola,
+    moolaMint,
+    simoleanIssuer,
+    simoleanMint,
+    simoleans,
+  } = setup();
+  const zoe = makeZoe(fakeVatAdmin);
+
+  // pack the contract
+  const bundle = await bundleSource(contractRoot);
+  // install the contract
+  const installation = await zoe.install(bundle);
+
+  // Alice creates an instance
+  const issuerKeywordRecord = harden({
+    Pixels: moolaIssuer,
+    Money: simoleanIssuer,
+  });
+
+  // This contract gives ZCF as the contractFacet for testing purposes
+  const { creatorFacet, publicFacet } = await E(zoe).startInstance(
+    installation,
+    issuerKeywordRecord,
+  );
+  /** @type ContractFacet */
+  const zcf = creatorFacet;
+
+  const seat1 = await zoe.offer(
+    publicFacet.makeInvitationForSeat(),
+    harden({ want: { A: moola(3) }, give: { B: simoleans(7) } }),
+    { B: simoleanMint.mintPayment(simoleans(7)) },
+  );
+  const seat2 = await zoe.offer(
+    publicFacet.makeInvitationForSeat(),
+    harden({ want: { B: simoleans(3) }, give: { A: moola(5) } }),
+    { A: moolaMint.mintPayment(moola(5)) },
+  );
+  const zcfSeat1 = await seat1.getOfferResult();
+  const zcfSeat2 = await seat2.getOfferResult();
```

Comment:
Rather than returning the zcfSeat as the offerResult, what do you think of making some helpers for tests that do it from the outside, like:

```js
  const makeInvitation = () => {
    let zcfSeat;
    const getSeat = seat => (zcfSeat = seat);
    return { zcfSeat, invitation: zcf.makeInvitation(getSeat, 'seat') };
  };

  const makeOffer = async (proposal, payments) => {
    const { zcfSeat, invitation } = makeInvitation();
    const userSeat = await E(zoe).offer(invitation, proposal, payments);
    return { zcfSeat, userSeat };
  };

  const seats1 = await makeOffer(
    harden({ want: { A: moola(3) }, give: { B: simoleans(7) } }),
    { B: simoleanMint.mintPayment(simoleans(7)) },
  );
  const seats2 = await makeOffer(
    harden({ want: { B: simoleans(3) }, give: { A: moola(5) } }),
    { A: moolaMint.mintPayment(moola(5)) },
  );
  const message = await swap(zcf, seats1.zcfSeat, seats2.zcfSeat);
  t.is(
    message,
    'The offer has been accepted. Once the contract has been completed, please check your payout',
  );
```

It's about the same amount of effort but seems to not be so tightly coupled to a particular contract and allows the offerResult to be used for something else.

Reply:
Oh whoops, as written this will not work. Let me amend

Reply:
I think this would work

```js
const makeOffer = async (zoe, zcf, proposal, payments) => {
  let zcfSeat;
  const getSeat = seat => {
    zcfSeat = seat;
  };
  const invitation = await zcf.makeInvitation(getSeat, 'seat');
  const userSeat = await E(zoe).offer(invitation, proposal, payments);
  return { zcfSeat, userSeat };
};
```

Reply:
yep, that worked.